integration_tests/src/main/python/json_fuzzing_test.py

```diff
@@ -0,0 +1,417 @@
+                yield t
+
+def gen_hex():
+    """
+    HEX -> DIGIT 
+         | 'a' .. 'f' 
+         | 'A' .. 'F'
+    """
+    path = random.randint(0, 2)
+    if path == 0:
+        for t in gen_digit():
+            yield t
+    elif path == 1:
+        yield chr(random.randint(0x41, 0x46))
+    else:
+        yield chr(random.randint(0x61, 0x66))
+
+def gen_number():
+    """
+    NUMBER -> INTEGER FRACTION EXPONENT
+    """
+    for t in gen_integer():
+        yield t 
+    for t in gen_fraction():
+        yield t 
+    for t in gen_exponent():
+        yield t 
+
+def gen_integer():
+    """
+    INTEGER -> DIGIT 
+             | ONENINE DIGITS 
+             | '-' DIGIT 
+             | '-' ONENINE DIGITS
+    """
+    if random.randint(1, 100) <= 50:
+        yield '-'
+    
+    if random.randint(1, 100) <= 50:
+        for t in gen_digit():
+            yield t 
+    else:
+        for t in gen_onenine():
+            yield t 
+        for t in gen_digits():
+            yield t 
+
+def gen_digits():
+    """
+    DIGITS -> DIGIT 
+            | DIGIT DIGITS
+    """
+    for t in gen_digit():
+        yield t 
+
+    if random.randint(1, 100) < 70:
+        for t in gen_digits():
+            yield t 
+
+def gen_digit():
+    """
+    DIGIT -> '0' 
+           | ONENINE
+    """
+    if random.randint(0, 9) == 0:
+        yield '0'
+    else:
+        for t in gen_onenine():
+            yield t
+
+def gen_onenine():
+    """
+    ONENINE -> '1' .. '9'
+    """
+    yield chr(random.randint(0x31, 0x39))
+
+def gen_fraction():
+    """
+    FRACTION -> "" | '.' DIGITS
+    """
+    if random.randint(1, 100) < 50:
+        yield ""
+    else:
+        yield '.'
+        for t in gen_digits():
+            yield t 
+
+def gen_exponent():
+    """
+    EXPONENT -> "" 
+              | 'E' SIGN DIGITS 
+              | 'e' SIGN DIGITS
+    """
+    if random.randint(1, 100) < 20:
+        yield ""
+    else:
+        yield random.choice(['E', 'e'])
+        for t in gen_sign():
+            yield t 
+        for t in gen_digits():
+            yield t 
+
+def gen_sign():
+    """
+    SIGN -> "" 
+          | '+' 
+          | '-'
+    """
+    yield random.choice(["", '+', '-'])
+
+def gen_whitespace():
+    """
+    WHITESPACE -> '' 
+                | 0x0020 WHITESPACE 
+                | 0x000a WHITESPACE (todo)
+                | 0x000d WHITESPACE (todo)
+                | 0x0009 WHITESPACE (todo)
+    """
+    if random.randint(0, 4) > 3:
+        yield chr(random.choice([0x0020]))
+        for t in gen_whitespace():
+            yield t
+    else:
+        yield ''
+
+def gen_bool():
+    """
+    BOOL -> "true" 
+          | "null" 
+          | "false"
+    """
+    yield random.choice(["true", "null", "false"])
+
+
+
+_enable_all_types_conf = {
+    'spark.rapids.sql.format.json.enabled': 'true',
+    'spark.rapids.sql.format.json.read.enabled': 'true'}
+
+@approximate_float
+@pytest.mark.xfail(reason = "fuzz test")
```

Comment:
should there be at least a second test that is always expected to succeed?

Naively I am also not sure why this test would be expected to fail. I guess the other point to make, would be to add a github issue link on what is failing. Perhaps that's well known to others, but it isn't obvious from this change to me.

Reply:
The aim of this fuzzy test is to help find more corner cases in JSON reading. Link: https://github.com/NVIDIA/spark-rapids/issues/4821

Initially, my thought was that it would spoil our CI tests if a corner case was found. So I marked it as `xfail`. However, after thinking carefully about your feedback, I remove the `xfail`. Maybe we should let the CI fail when we find a new corner case, although the possibility is low.

Reply:
Done